tests/src/ut_test_kvp.c

```diff
@@ -587,6 +593,303 @@ void test_ut_kvp_bool(void)
+
+    result = ut_kvp_fieldPresent( gpMainTestInstance, "6" );
+    UT_ASSERT( result == false );
+
+    /* Positive Tests */
+    result = ut_kvp_fieldPresent( gpMainTestInstance, "1/value" );
+    UT_ASSERT( result == true );
+
+    result = ut_kvp_fieldPresent( gpMainTestInstance, "2" );
+    UT_ASSERT( result == true );
+
+    result = ut_kvp_fieldPresent( gpMainTestInstance, "3/value" );
+    UT_ASSERT( result == true );
+
+    result = ut_kvp_fieldPresent( gpMainTestInstance, "4.value" );
+    UT_ASSERT( result == true );
+
+    result = ut_kvp_fieldPresent( gpMainTestInstance, "5.value" );
+    UT_ASSERT( result == true );
+
+    result = ut_kvp_fieldPresent( gpMainTestInstance, "5" );
+    UT_ASSERT( result == true );
+
+}
+
+void test_ut_kvp_open_singleIncludeFileWithBuildFromFile( void )
+{
+    ut_kvp_instance_t *pInstance = NULL;
+    ut_kvp_status_t status;
+    char* kvpData;
+
+    pInstance = ut_kvp_createInstance();
+    if ( pInstance == NULL )
+    {
+        assert( pInstance != NULL );
+        UT_LOG_ERROR("ut_kvp_open() - Read Failure");
+        return;
+    }
+
+    status = ut_kvp_open( pInstance, KVP_VALID_TEST_SINGLE_INCLUDE_FILE_YAML);
+    assert( status == UT_KVP_STATUS_SUCCESS );
+
+    if ( status != UT_KVP_STATUS_SUCCESS )
+    {
+        UT_LOG_ERROR("ut_kvp_open() - Read Failure");
+        return;
+    }
+
+    kvpData = ut_kvp_getData(pInstance);
+
+    if(kvpData != NULL)
+    {
+        // Print the emitted KVP string
+        printf("%s\n", kvpData);
+
+        // Free the emitted KVP string
+           free(kvpData);
+    }
+
+     ut_kvp_destroyInstance( pInstance );
+
+}
+
+void test_ut_kvp_singleIncludeUrlsWithBuildFromFile( void )
+{
+    ut_kvp_instance_t *pInstance = NULL;
+    ut_kvp_status_t status;
+    char* kvpData;
+
+    pInstance = ut_kvp_createInstance();
+    if ( pInstance == NULL )
+    {
+        assert( pInstance != NULL );
+        UT_LOG_ERROR("ut_kvp_open() - Read Failure");
+        return;
+    }
+
+    status = ut_kvp_open( pInstance, KVP_VALID_TEST_SINGLE_INCLUDE_URL_YAML);
+    assert( status == UT_KVP_STATUS_SUCCESS );
+
+    if ( status != UT_KVP_STATUS_SUCCESS )
+    {
+        UT_LOG_ERROR("ut_kvp_open() - Read Failure");
+        return;
+    }
+
+    kvpData = ut_kvp_getData(pInstance);
+
+    if(kvpData != NULL)
+    {
+        // Print the emitted KVP string
+        printf("%s\n", kvpData);
+
+        // Free the emitted KVP string
+           free(kvpData);
+    }
+
+     ut_kvp_destroyInstance( pInstance );
+
+}
+
+void test_ut_kvp_IncludeDepthCheckWithBuildFromFile(void)
+{
+    ut_kvp_instance_t *pInstance = NULL;
+    ut_kvp_status_t status;
+    char* kvpData;
+
+    pInstance = ut_kvp_createInstance();
+    if ( pInstance == NULL )
+    {
+        assert( pInstance != NULL );
+        UT_LOG_ERROR("ut_kvp_open() - Read Failure");
+        return;
+    }
+
+    status = ut_kvp_open( pInstance, KVP_VALID_TEST_DEPTH_CHECK_INCLUDE_YAML);
```

Comment:
looks same case as previous two.

create a common function and call it from all of the tests required, and pass params.